Backend/src/main/java/dev/learning/fahabookstore/service/BookServiceImplement.java

```diff
@@ -0,0 +1,52 @@
+package dev.learning.fahabookstore.service;
+
+import dev.learning.fahabookstore.entity.Book;
+import dev.learning.fahabookstore.exception.ResourceNotFoundException;
+import dev.learning.fahabookstore.repository.BookRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class BookServiceImplement implements BookService {
+    private BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImplement(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    @Override
+    @Transactional
+    public Book save(Book theBook) {
+        return bookRepository.save(theBook);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return bookRepository.findAll();
+    }
+
+    @Override
+    public Book findById(int theId) {
+        Optional<Book> result = bookRepository.findById(theId);
```

Comment:
you can use method of Optional class for check null or throw Exception for short